### PR TITLE
CLOUDP-69289: Fix custom StatefulSet spec merging of Env Vars

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

Our arbitrary StatefulSet Spec merging was not working correctly when attempting to add additional Environment Variables, this PR fixes this issue.

### CODE REVIEW

File: pkg/kube/container/containers.go [hunk 1]
```diff
@@ -103,31 +105,10 @@ func WithLifecycle(lifeCycleMod lifecycle.Modification) Modification {
 // WithEnvs ensures all of the provided envs exist in the container
 func WithEnvs(envs ...corev1.EnvVar) Modification {
 	return func(container *corev1.Container) {
-		container.Env = mergeEnvs(container.Env, envs)
+		container.Env = envvar.Merge(container.Env, envs)
 	}
 }
 
-func mergeEnvs(existing, desired []corev1.EnvVar) []corev1.EnvVar {
```
Comment (chatton) [inline pkg/kube/container/containers.go:110]: moved to separate package as this is now being used in multiple places

File: pkg/apis/mongodb/v1/mongodb_types.go [hunk 2]
```diff
@@ -47,10 +48,10 @@ type MongoDBSpec struct {
 
 	// Users specifies the MongoDB users that should be configured in your deployment
 	// +required
-	Users []MongoDBUser `json:"users"`
+	Users []MongoDBUser `json:"-"` // `json:"users"`
```
Comment (chatton) [inline pkg/apis/mongodb/v1/mongodb_types.go:51]: don't include in CRD until SCRAM feature is complete

File: pkg/util/envvar/envvars.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package envvar
+
+import (
+	"sort"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func Merge(existing, desired []corev1.EnvVar) []corev1.EnvVar {
```
Comment (LouisPlisso) [inline pkg/util/envvar/envvars.go:9]: do we want to call it MergeWithOverride to be explicit?

Comment (chatton) [reply pkg/util/envvar/envvars.go:9]: good idea 👍

**Pull request merged**